### PR TITLE
Add ability to parse raw text into docvectors on-the-fly for impact indexes #2122

### PR DESCRIPTION
**Pull request by AileenLin**
#2122

### CODE REVIEW

File: src/main/java/io/anserini/search/SimpleImpactSearcher.java [hunk 1]
```diff
@@ -143,6 +192,136 @@ private boolean empty_encoder(){
...
+  /**
+   * Returns the analyzer used.
+   *
+   * @return analyzed used
+   */
+  public Analyzer get_analyzer(){
+    return this.analyzer;
+  }
+
+    /**
```
Comment (lintool) [inline src/main/java/io/anserini/search/SimpleImpactSearcher.java:213]:
```suggestion
  /**
```

File: src/main/java/io/anserini/search/SimpleImpactSearcher.java [hunk 2]
```diff
@@ -143,6 +192,136 @@ private boolean empty_encoder(){
...
+  /**
+   * Disables RM3 query expansion.
+   */
+  public void unset_rm3() {
+    this.useRM3 = false;
+    cascade = new RerankerCascade();
+    cascade.add(new ScoreTiesAdjusterReranker());
+  }
+
+    /**
```
Comment (lintool) [inline src/main/java/io/anserini/search/SimpleImpactSearcher.java:231]: fix indent, same as above.

**Review by lintool — COMMENTED** [review]
I thought we were going to change type to `Map<String, Int>` in places where it makes sense?

File: src/main/java/io/anserini/search/SimpleImpactSearcher.java [hunk 3]
```diff
@@ -429,48 +501,47 @@ public Map<String, Float> encode_with_onnx(String queryString) throws OrtExcepti
...
    * @param k number of hits
    * @return array of search results
    * @throws IOException if error encountered during search
+   * @throws OrtException if error encountered during search
    */
-  public Result[] search(Map<String, Float> q, int k) throws IOException, OrtException {
-    Query query = generator.buildQuery(Constants.CONTENTS, q);
-    String encodedQuery = encode_with_onnx(q);
+  public Result[] search(Map<String, Integer> encoded_q, int k) throws IOException, OrtException {
+    Map<String, Float> float_encoded_q = intToFloat(encoded_q);
```
Comment (lintool) [inline src/main/java/io/anserini/search/SimpleImpactSearcher.java:542]: I'm still not very happy with this... we get a `Map<String, Integer>`... and then we covert to `Map<String, Float>`... then it comes back as a flat String.

Should we be redesigning the QueryEncoder?

Comment (AileenLin) [reply src/main/java/io/anserini/search/SimpleImpactSearcher.java:542]: just saw this. this line is not for query encoder thing, its for QueryGenerator to build query, which takes in float weight. so here I just convert Int to Float, for example 10 -> 10.0

File: src/main/java/io/anserini/search/SimpleImpactSearcher.java [hunk 4]
```diff
@@ -545,6 +625,26 @@ public Result[] search(Map<String, Integer> encoded_q, int k) throws IOException
...
+   *
+   * @param q string query
+   * @param k number of hits
+   * @return array of search results
+   * @throws IOException if error encountered during search
+   * @throws OrtException if error encountered during search
+   */
+  public Result[] search(String q, int k) throws IOException, OrtException {
+    // make encoded query from raw query
+    Map<String, Integer> encoded_q = encode_with_onnx(q);
```
Comment (lintool) [inline src/main/java/io/anserini/search/SimpleImpactSearcher.java:639]: Same, camelCase?

**Review by lintool — APPROVED** [review]

**Pull request merged**